vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -156,7 +156,10 @@ function Disable-WindowsUpdates {
 function Get-ContainerImages {
     Write-Log "Pulling images for windows server $windowsSKU" # The variable $windowsSKU will be "2019-containerd", "2022-containerd", ...
     foreach ($image in $imagesToPull) {
-        if (($image.Contains("mcr.microsoft.com/windows/servercore") -and ![string]::IsNullOrEmpty($env:WindowsServerCoreImageURL)) -or
+        # If the target image to pull contains the prefix "mcr.microsoft.com/windows/servercore" or "mcr.microsoft.com/windows/nanoserver",
+        # you need to you should add it to exception list to avoid using $env:WindowsServerCoreImageURL or $env:WindowsNanoServerImageURL.
+        # Example: "mcr.microsoft.com/windows/servercore/iis:latest"
+        if (($image.Contains("mcr.microsoft.com/windows/servercore") -and ![string]::IsNullOrEmpty($env:WindowsServerCoreImageURL) -and !$image.Contains("iis")) -or
```

Comment:
We will hit the same problem if we introduce another image similar to iis in future.
Suggest to split $image with `:` and then compare it

Reply:
Thanks for the comment. Done.